### DISCUSSION TITLE
Regression in reStructuredText (rst) handling

### DISCUSSION THREAD

**Discussion by ale5000-git** [community/community#121163] [Repositories]
### Select Topic Area

Bug

### Body

Hi,
GitHub cannot display correctly tables in reStructuredText; they was working fine in the past so it is a regession.
Example: https://github.com/ale5000-git/RSTtest/blob/main/README.rst

Best regards,
ale5000

**Comment by evdcush** [comment]
I can confirm this regression.

GitHub now incorrectly renders a valid literal/code/pre-formatted block of text.

Example: 
https://github.com/evdcush/fart/blob/master/README.rst


GitHub rST Regression             |  Correct Rendering
:-------------------------:|:-------------------------:
![screencap_2024-05-11_18-40-25](https://github.com/community/community/assets/28853935/f3af0340-d38a-48db-a979-c88ea05cacf8)  |  ![screencap_2024-05-11_18-39-30](https://github.com/community/community/assets/28853935/2e54f126-6ea7-4dba-afe1-6c3b246d1570)


Please revert whatever recent changes were made to rendering reStructuredText.

**Comment by evdcush** [comment]
Regression persists; please fix.

**Comment by evdcush** [comment]
Still broken.

**Reply by evdcush** [reply]
rST regression persists.

**Comment by evdcush** [comment]
Same bug reported (and unanswered/unaddressed) here:


https://github.com/orgs/community/discussions/86715

**Comment by dargueta** [comment]
I see that a lot of people are having trouble with large files. Mine is only 200 lines, and yet some (but not all) tables are very broken ([permalink](https://github.com/dargueta/libmetaldata/blob/292a8d36db749f2f696f2ce20c269cb96bbd7651/README.rst)). They rendered fine before. The following examples are taken from the same file, and are the only tables. Screenshots of the correctly rendered code are taken from [CLion](https://www.jetbrains.com/clion/)'s RST previewer.

Hopefully this additional information helps!

# Simple syntax, works

## Code

```rst
========= ========= ==================
What      Header    Unhosted Fallbacks
========= ========= ==================
bool      stdbool.h \*
free      stdlib.h  N
malloc    stdlib.h  N
memcmp    string.h  Y
memcpy    string.h  Y
memset    string.h  Y
realloc   stdlib.h  N
SIZE_MAX  stdint.h  \*
size_t    stddef.h  \*
strcmp    string.h  Y
========= ========= ==================
```

## Renders  as

![rst-simple-ok](https://github.com/community/community/assets/620513/1753f71a-84c7-4ad8-926c-d7804bde45e3)

# Simple syntax, broken

## The code

```rst
====================== ======== ============
Architecture           Compiles Passes Tests
====================== ======== ============
AMD64 (x86-64)         Y        Y
DS390                  Y
DS400                  Y
ez80-z80               Y
HC08 (Motorola 68HC08) Y
macOS ARM64            Y        Y
MCS-51                 Y
MOS 6502               Y
MOS 65C02              Y
Padauk PDK 13          Y
Padauk PDK 14          Y
Padauk PDK 15          Y
Rabbit 800             Y
Rabbit 2000            Y
Rabbit 2000A           NS
Rabbit 3000A           Y
S08                    Y
SM83                   NS
STM8                   Y
TININative             Y [#]_
TLCS 90                Y
x86-32                 Y        Y
z180                   Y
z80                    Y
z80n                   NS
====================== ======== ============
```

## Renders as

![Screenshot from 2024-05-26 06-27-02](https://github.com/community/community/assets/620513/e77c6705-41d8-48c9-9168-a8133785dae0)

## Expected

(The full table is longer but my screen isn't big enough to show it all at once.)

![Screenshot from 2024-05-26 06-34-34](https://github.com/community/community/assets/620513/a70f13a4-5c4e-49fd-bf9f-dfe67c3d2a9f)


# Complex syntax

## The code

```rst
+-----------------+----------+-----------------------+-----------------------+
| Compiler        | OS       | Hosted                | Unhosted              |
|                 |          +----------+------------+----------+------------+
|                 |          | Compiles | Tests Pass | Compiles | Tests Pass |
+=================+==========+==========+============+==========+============+
| Clang_          | Cygwin   | Y        | Y          | Y        | Y          |
|                 +----------+----------+------------+----------+------------+
|                 | Linux    | Y        | Y          | Y        | Y          |
|                 +----------+----------+------------+----------+------------+
|                 | MacOS    | Y        | Y          | Y        | Y          |
+-----------------+----------+----------+------------+----------+------------+
| GCC_            | Cygwin   | Y        | Y          | Y        | Y          |
|                 +----------+----------+------------+----------+------------+
|                 | Linux    | Y        | Y          | Y        | Y          |
|                 +----------+----------+------------+----------+------------+
|                 | MacOS    | Y        | Y          | Y        | Y          |
+-----------------+----------+----------+------------+----------+------------+
| Intel           | MacOS    |          |            |          |            |
|                 +----------+----------+------------+----------+------------+
|                 | Linux    |          |            |          |            |
+-----------------+----------+----------+------------+----------+------------+
| MinGW_          | Cygwin   | Y        | Y          | Y        | Y          |
|                 +----------+----------+------------+----------+------------+
|                 | Linux    | Y [#]_   | NS         | Y        | NS         |
+-----------------+----------+----------+------------+----------+------------+
| OpenWatcom      | MacOS    |          |            |          |            |
|                 +----------+----------+------------+----------+------------+
|                 | Linux    |          |            |          |            |
+-----------------+----------+----------+------------+----------+------------+
| SDCC_           | See below for the list of architectures this compiles for|
+-----------------+----------+----------+------------+----------+------------+
| TCC_            | Linux    | Y        | Y          | Y        | Y          |
+-----------------+----------+----------+------------+----------+------------+
| Visual Studio   | Windows  |          |            |          |            |
+-----------------+----------+----------+------------+----------+------------+
```

## Renders as

![Screenshot from 2024-05-26 06-29-14](https://github.com/community/community/assets/620513/7082077d-8f26-4073-ab64-ca25c1b99863)

## Expected

![Screenshot from 2024-05-26 06-33-10](https://github.com/community/community/assets/620513/9b329e8b-828e-41a8-8e37-e5a7a7fe6b78)

**Comment by evdcush** [comment]
Bug persists.

When will GitHub acknowledge the bug?

**Reply by evdcush** [reply]
Issue persists.

**Reply by evdcush** [reply]
Issue persists.

**Comment by evdcush** [comment]
The bug was recently patched!

Reported in the other rST regression discussion:

- https://github.com/orgs/community/discussions/86715#discussioncomment-9705928

I'd like to know the root cause of the regression though 🤔 

@fireattack's comment seems plausible:

https://github.com/orgs/community/discussions/86715#discussioncomment-9706063
